src/igmpproxy.h

```diff
@@ -144,6 +144,7 @@ void my_log( int Serverity, int Errno, const char *FmtSt, ... );
 struct SubnetList {
     uint32_t            subnet_addr;
     uint32_t            subnet_mask;
+    unsigned short      allow;
     struct SubnetList   *next;
 };
```

Comment:
Put `allow` at the end of structure. On 64-bit systems, compiler has to align `next` pointer to be on 64-bit offset from beginning of structure, therefore it will change type of `allow` member from 16-bit (short) to 64-bit integer type.

Also igmpproxy already requires C99, so you can use `bool` type without any issue.

Reply:
Forgot about allignment. Thanks!

`allow` has been moved and changed to `bool`,